src/oauth-manager/oauth-utils.ts

```diff
@@ -423,6 +423,11 @@ export function validateAndSanitizeUrl(url: string): string {
     // Remove any whitespace
     const trimmedUrl = url.trim();
 
+    // Reject 'null' as an invalid URL
+    if (trimmedUrl.toLowerCase() === 'null') {
+      throw new Error('Invalid URL string.');
```

Comment:
McpServerError ?

Also, when will this happen? If the `trimmerUrl` is null then line 424 will break.